src/include/main/db_main.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <network/terrier_server.h>
+#include <memory>
+#include <unordered_map>
+#include <utility>
+#include "catalog/catalog.h"
+#include "common/action_context.h"
+#include "common/stat_registry.h"
+#include "settings/settings_param.h"
+#include "transaction/transaction_manager.h"
+
+namespace terrier {
+
+namespace settings {
+class SettingsManager;
+class SettingsTests;
+}  // namespace settings
+
+/**
+ * The DBMain Class holds all the static functions, singleton pointers, etc.
```

Comment:
I think we should mention that the db main serves as a global context of the whole DBMS in the documentation.